core/src/message_queue.c

```diff
@@ -24,15 +24,20 @@ typedef struct MESSAGE_QUEUE_TAG
 static MESSAGE_HANDLE message_pop(MESSAGE_QUEUE_HANDLE_DATA* handle)
 {
     MESSAGE_HANDLE result;
-    MESSAGE_QUEUE_STORAGE* entry = 
+	if (DList_IsListEmpty((PDLIST_ENTRY)&(handle->queue_head)))
```

Comment:
>if (DList_IsListEmpty((PDLIST_ENTRY)&(handle->queue_head))) [](start = 1, length = 59)

Don't you need to test first if handle is null before dereferencing? (Do you have a test for message_pop where handle is null?